### ISSUE TITLE
a bug of split layer

### ISSUE THREAD

**Issue by shixinlishixinli** [openvinotoolkit/openvino#4674]
##### System information (version)

- OpenVINO => 2021.1
- Operating System / Platform => ubuntu 1804
- Problem classification: IE inference
- Framework: paddle (if applicable)
- Model name: paddle ocr (if applicable)

##### Detailed description

paddle ocr is a powerful ocr model . i convert model form paddle  to onnx  , and the onnx model can work in onnxruntime successfully . then, i try to use IE inference this model , there is a bug of split layer.

File "ie_api.pyx", line 311, in openvino.inference_engine.ie_api.IECore.load_network
  File "ie_api.pyx", line 320, in openvino.inference_engine.ie_api.IECore.load_network
RuntimeError: AssertionFailed: layer->get_output_partial_shape(i).is_static() nGraph Split operation with name: Slice@13/Output_0/Data_/DecomposedBiLSTM_0 cannot be converted to Split layer with name: Slice@13/Output_0/Data_/DecomposedBiLSTM_0 because output with index 0 contains dynamic shapes: ?. Try to use CNNNetwork::reshape() method in order to specialize shapes before the conversion.

         <layer id="476" name="Slice@13/Output_0/Data_/DecomposedBiLSTM_0" type="Split" version="opset1">
 7874             <data num_splits="2"/>
 7875             <input>
 7876                 <port id="0">
 7877                     <dim>2</dim>
 7878                     <dim>1</dim>
 7879                     <dim>48</dim>
 7880                 </port>
 7881                 <port id="1">
 7882                 </port>
 7883             </input>
 7884             <output>
 7885                 <port id="2" precision="FP32">
 7886                     <dim>1</dim>
 7887                     <dim>1</dim>
 7888                     <dim>48</dim>
 7889                 </port>
 7890                 <port id="3" precision="FP32">
 7891                     <dim>1</dim>
 7892                     <dim>1</dim>
 7893                     <dim>48</dim>
 7894                 </port>
 7895             </output>
 7896         </layer>

 
there is no dynamic shapes in split layer , but the ie.load_network has a bug. so how did i solve it 

i am waitting for your reply
Best
Lisa Shi
##### Issue submission checklist

 - [ ] I report the issue, it's not a question
   
 - [ ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ] There is reproducer code and related data files: images, videos, models, etc.

**Comment by Iffa-Meah** [comment]
Hi,

Any chance that you convert your model to IR?
You won't be able to use OpenVINO with the original model format, you need the IR.

[This](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_Convert_Model_From_ONNX.html) might help you if you are using ONNX topologies as listed in the documentation.


Fyi,  If you are trying to convert **Keras** model to IR, you will need to refer [here](https://github.com/onnx/keras-onnx).

**Comment by shixinlishixinli** [comment]
yes i have converted model form onnx to openvino by MO successfully, and the out.zip include bin file and xml file of the model . 
[out.zip](https://github.com/openvinotoolkit/openvino/files/6099643/out.zip)

**Comment by Iffa-Meah** [comment]
Is it possible for you to share your model source? the paddle and also onnx.
Plus, also your commands to convert them.

**Comment by shixinlishixinli** [comment]
this is paddle ocr model source https://github.com/PaddlePaddle/PaddleOCR
there are three models in paddleOCR, det, cls and rec. i have successfully convert det and cls from paddle to openvino. and the inference result is correct in openvino

i use paddle2onnx to covert model form paddle to onnx. the command is "paddle2onnx --model_dir . --model_filename ./inference.pdmodel  --params_filename ./inference.pdiparams --save_file onnx.onnx --opset_version 11"
paddle2onnx need to be installed by source of github . the latest paddle2onnx can successfully convert paddleocr form paddle to onnx

the onnxtuntime source can be found in https://aistudio.baidu.com/aistudio/projectdetail/1479970
and i test the rec.onnx by onnxruntime the result is right. rec.onnx can be converted to out.xml and out.bin by MO. the out.zip include the out.xml and out.bin

**Comment by ilya-lavrenov** [comment]
CC @slyalin

**Comment by slyalin** [comment]
I've tried to run the model from out.zip archive with benchmark_app sample, it fails with a different message saying that
`While validating node 'v1::Split Split_1684 (Slice@13[0]:f32{?,?,?}, 5892_const[0]:i64{1}) -> ()' with friendly_name 'Split_1684':
The 'axis' input is expected to be a scalar. Got: {1}`. For some reason it looks running through this place correctly in your OpenVINO 2021.1. I'm using some branch close to master branch. So IR doesn't look correct. Could you try to use more up-to-date version of OpenVINO to try to convert from ONNX to IR? If you have to stick with 2021.1, I will switch to that release branch, it will require some time. Just to be on the same page.

But anyway it seems that even if we align versions etc., the problem is in ScatterNDUpdate which doesn't have evaluate method defined to calculate shape for Broadcast op that is fed by StridedSlice that finally goes to our problematic Split. That's why it says Slice@13[0]:f32{?,?,?} with ? marks -- it cannot deduce shape because of missing ScatterNDUpdate::evaluate.

You can try to quickly solve it by disabling model reshapability as ScatterNDUpdate is left in the model because ModelOptimizer generates reshapable IRs by default and that operation is actually a part of shape calculation sub-graph. Please try using flag --static_shape when converting onnx model to IR.

Please let me know if model reshapability in run-time is something desired and you cannot really use that method with --static_shape in long term.
